code/modules/modular_computers/file_system/catalog.dm

```diff
@@ -0,0 +1,331 @@
+/*
+	important notes
+	catalogs are handled in /datum/nano_module, check there
+	important procs are:
+		browse_catalog_entry()
+		browse_catalog()
+		refresh_catalog_browsing()
+
+	TODO: add access level that will show more info
+*/
+/datum
+	var/catalog_info_level_one
+	var/catalog_info_level_two
+	var/catalog_info_level_three
+	var/catalog_info_level_four
+	var/catalog_info_level_ooc
+
+/atom
+	// should this atom have catalog entry
+	// this is set to FALSE for not gameplay atoms like areas, landmarks etc
+	var/contribute_to_catalog = TRUE
+	// this will add atom to can_be_found reagent entry
+	// this is set to FALSE for things like beakers and pills 
+	var/contribute_to_reagent_catalog = TRUE
+	// this will add atom to can_be_found storage item enry
+	var/contribute_to_container_catalog = TRUE
+	// will create icon asset that will be send to players
+	var/create_icon_asset = TRUE
+
+GLOBAL_LIST_EMPTY(catalogs)
+GLOBAL_LIST_EMPTY(all_catalog_entries_by_type)
+
+/hook/startup/proc/createCatalogs()
```

Comment:
It's better to use subsystem instead of hooks. They're much more controllable.

Reply:
I did a subsystem for creating catalogs entries for atoms, but i was had to remove it in this PR.
I need reagent entries and probably some other datums to be created before atoms starts initializing so it should be here or create additional subsystem for stuff like that.

This will probably be improved in the future so ill live it so for now

Reply:
> I need reagent entries and probably some other datums to be created before atoms starts initializing so it should be here or create additional subsystem for stuff like that.

Sounds simple. You can set higher priority for Codex subsystem to initialize it before atoms or vise versa.
Also there exist think like `PreInit(..)`